common/cli/factory.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/cadence/.gen/go/cadence/workflowserviceclient"
+	"go.uber.org/cadence/client"
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/transport/tchannel"
+	"go.uber.org/zap"
+)
+
+const (
+	_cadenceClientName      = "cadence-client"
+	_cadenceFrontendService = "cadence-frontend"
+)
+
+// Environment is the cadence environment
+type Environment string
+
+const (
+	// Development is for laptop or development testing
+	Development Environment = "development"
+	// Staging is the cadence staging cluster
+	Staging Environment = "staging"
+	// Production is the cadence production cluster
+	Production Environment = "prod"
+)
+
+// ConvertEnv takes a string and convert to type Environment
+func ConvertEnv(env string) Environment {
+	env = strings.ToLower(env)
+	if env == "staging" {
+		return Staging
+	}
+	if env == "production" {
+		return Production
+	}
+	return Development
+}
+
+// WorkflowClientBuilderInterface is an interface to build client to cadence service
+type WorkflowClientBuilderInterface interface {
```

Comment:
Why is this file here? Where is it being used? Can't find any references in the diff.

Reply:
I previously thought common might be a good place, and later playground or other apps can use it. But now I think it might be better just put everything in cli folder for now, so changed. 
This interface is used in "commands.go", and I need it so that I can have a customized builder used internal. See update comments in the file.